apiserversdk/proxy.go

```diff
@@ -84,3 +92,89 @@ func requireKubeRayService(handler http.Handler, k8sClient *kubernetes.Clientset
 		handler.ServeHTTP(w, r)
 	})
 }
+
+type retryRoundTripper struct {
+	base    http.RoundTripper
+	retries int
+}
+
+func newRetryRoundTripper(base http.RoundTripper, retries int) http.RoundTripper {
+	return &retryRoundTripper{base: base, retries: retries}
+}
+
+func (rrt *retryRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	timeoutCtx, cancel := context.WithTimeout(req.Context(), 30*time.Second)
```

Comment:
two comments here:
- please make overall retry timeout a constant and compatible with v1
- I'm not sure what's the different between "timeout" and "max attempt" here, to me both are retry configs

Reply:
If `RoundTrip` cannot take a context, I would suggest to pass context into the struct.
Because the real timeout should be the earliest deadline